rendermodules/point_match_optimization/pt_match_optimization.py

```diff
@@ -0,0 +1,583 @@
+    "tile_stack": "point_match_optimization_test",
+    "stack": "em_2d_montage_apply_mipmaps",
+    "tilepair_file": "/allen/programs/celltypes/workgroups/em-connectomics/gayathrim/scratch/17797_1R/tilepairs/temca4_tilepair_test.json",
+    "filter_tilepairs":False,
+    "no_tilepairs_to_test":4,
+    "max_tilepairs_with_matches":4  
+}
+
+
+def polys_overlap(ts, buffer=0):
+    ts1 = ts[0]
+    ts2 = ts[1]
+    p1 = Polygon(ts1.bbox_transformed(ndiv_inner=1))
+    p2 = Polygon(ts2.bbox_transformed(ndiv_inner=1))
+    return p1.intersects(p2)
+
+
+'''
+def get_canvas_url(render_args, stack, tile1, tile2, url_options):
+    base_data_url = '%s:%d/render-ws/v1'%(render_args['host'], render_args['port'])
+    tile_base_url = "%s/owner/%s/project/%s/stack/%s/tile"%(base_data_url,
+                                                            render_args['owner'],
+                                                            render_args['project'],
+                                                            stack)
+    url_suffix = "render-parameters"
+    if url_options['renderWithFilter'] is False:
+        url_suffix += '?filter=false'
+    else:
+        url_suffix += '?filter=true'
+    if url_options['normalizeForMatching'] is False:
+        url_suffix += '&normalizeForMatching=false'
+    else:
+        url_suffix += '&normalizeForMatching=true'
+    if url_options['renderWithoutMask'] is False:
+        url_suffix += '&renderWithoutMask=false'
+    else:
+        url_suffix += '&renderWithoutMask=true'
+
+    canvas_urls = "%s/%s/%s %s/%s/%s"%(tile_base_url,
+                                        tile1,
+                                        url_suffix,
+                                        tile_base_url,
+                                        tile2,
+                                        url_suffix)
+    return canvas_urls
+'''
+
+def get_parameter_sets_and_strings(SIFT_options):
+    #length = [len(x) for x in SIFT_options]
+    #maxlen = np.max(length)
+
+    new_options = {}
+    all_parameter_list = []
+    for key in SIFT_options:
+        if (~isinstance(SIFT_options[key], list)):
+            new_options[key] = list(SIFT_options[key])
+        else:
+            new_options[key] = SIFT_options[key]
+
+    keys = []
+    for n in new_options:
+        keys.append(n)
+        all_parameter_list.append(new_options[n])
+
+    all_combinations = list(itertools.product(*all_parameter_list))
+    #return all_parameter_list
+    return keys, all_combinations
+
+def render_from_template(directory, template_name, **kwargs):
+    loader = FileSystemLoader(directory)
+    env = Environment(loader=loader)
+    template = env.get_template(template_name)
+    return template.render(**kwargs)
+
+def draw_matches(render, stack, tile1, tile2, ptmatches, scale, outdir, color=None):
+    
+    im1_url = "%s:%d/render-ws/v1/owner/%s/project/%s/stack/%s/tile/%s/png-image?scale=%f&filter=false&normalizeForMatchin=true"%(render.DEFAULT_HOST,
```

Comment:
small typo in normalizeForMatching here and below -- but it seems like these should also match the url params given in the input json rather than a default.

Reply:
agreed